otel.go

```diff
@@ -55,42 +65,49 @@ func (lk *logkeeper) initOtel(ctx context.Context) error {
 		grip.Error(errors.Wrap(err, "otel error"))
 	}))
 
-	lk.tracer = tp.Tracer(packageName)
-
-	lk.closers = append(lk.closers, closerOp{
+	closers = append(closers, closerOp{
 		name: "tracer provider shutdown",
 		closerFn: func(ctx context.Context) error {
 			catcher := grip.NewBasicCatcher()
 			catcher.Wrap(tp.Shutdown(ctx), "trace provider shutdown")
 			catcher.Wrap(traceExporter.Shutdown(ctx), "trace exporter shutdown")
-			catcher.Wrap(lk.otelGrpcConn.Close(), "closing gRPC connection")
-
 			return catcher.Resolve()
 		},
 	})
+	useCustomName = true
+}
 
-	return nil
+func initTracer(name string) otelTrace.Tracer {
+	var tracerName string
+	if useCustomName {
+		tracerName = fmt.Sprintf(packageName, name)
+	} else {
+		tracerName = defaultName
+	}
+	return otel.GetTracerProvider().Tracer(tracerName)
```

Comment:
The truth is, I think it works to have 
```go
var tracer = otel.GetTracerProvider().Tracer(<tracerName>)
```
At the top of a file. iirc the tracer will be initialized when the tracer provider is later initialized.

If this is true it could simplify things.

Reply:
https://github.com/open-telemetry/opentelemetry-go/blob/main/internal/global/state.go#L42-L54

`GetTracerProvider` returns the default trace provider. In `LoadTraceProvider` we are explicitly setting the default trace provider, so we definitely want to init the tracer after `LoadTraceProvider`.

Reply:
But I think it maybe the way it works is that it returns a pointer to the trace provider (actually an interface) that then points to the initialized trace provider.

It worked for us in evergreen ([here](https://github.com/evergreen-ci/evergreen/blob/6d2d2cd8c71da036e032bd8714f397c791d5b89d/model/otel.go#L12) is an example).